Models/Consumable.cs

```diff
@@ -1,79 +1,80 @@
 ﻿using System;
-using System.IO;
+using System.Runtime.CompilerServices;
 using Microsoft.Xna.Framework;
 using Microsoft.Xna.Framework.Graphics;
 using ReLogic.Content;
+using Terraria;
+using Terraria.ID;
+using Terraria.Localization;
 using Terraria.ModLoader;
 
 namespace Munchies.Models {
 	public class Consumable {
 		public string DisplayText;
 		//public string TexturePath;
 		public Asset<Texture2D> Texture;
-		public (float X, float Y) AssetDimensions;
+		//public (float X, float Y) AssetDimensions;
 		public string HoverText;
 		public ConsumableType? Type;
 		public Color? CustomTextColor;
-		public Func<bool> HasBeenConsumed;
+		public Func<int> CurrentCount; // How many of the item have been consumed; for bools (aka single use consumables), this is either 0 or 1
+		public Func<int> TotalCount; // Total number of items that can be consumed; for bools, this is 1
 
 		public bool UsingMissingTexture;
 
-		public Consumable(string displayText, string texturePath, (float X, float Y) assetDimensions, string hoverText, ConsumableType type, Func<bool> hasBeenConsumed) {
-			DisplayText = displayText;
-			//TexturePath = texturePath;
-			//AssetDimensions = assetDimensions;
-			HoverText = hoverText;
-			Type = type;
-			CustomTextColor = null;
-			HasBeenConsumed = hasBeenConsumed;
+		#region Init
+		public Consumable(ModItem modItem, object CategoryOrCustomColor, Func<int> currentCount, Func<int> totalCount) {
+			DisplayText = modItem.DisplayName.ToString();
+			HoverText = modItem.Tooltip.ToString();
```

Comment:
This is not solving the localization problem outlined in the issue. Invoking .Tostring() or .Value fetches the string for the currently selected language. Changing language then won't change these until next mod reload. To remedy this, don't call these methods, and instead store DisplayText/HoverText as a `LocalizedText`